### PR TITLE
Add new iree-dialects LLVM project.

### PR DESCRIPTION
**Pull request by stellaraccident**
* Has a single iree_public dialect (with a dummy op).
* Has the boiler-plate for python and CAPI and a test.
* Not yet wired in/building anywhere.
* No bazel build yet.
* Left a number of TODOs for future me to fix upstream.

### CODE REVIEW

File: llvm-projects/iree-dialects/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,61 @@
+if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
+message(FATAL_ERROR
+  "This project is intended to be built as part of LLVM via "
+  "-DLLVM_EXTERNAL_PROJECTS=iree_llvm_sandbox "
```
Comment (benvanik) [inline llvm-projects/iree-dialects/CMakeLists.txt:4]: outdated?

File: llvm-projects/iree-dialects/lib/Dialects/iree_public/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  IREEPublicDialect.cpp
+  IREEPublicOps.cpp
+
+  ADDITIONAL_HEADER_DIRS
+  ${IREE_DIALECTS_SOURCE_DIR}/include
+
+  DEPENDS
+  MLIRIREEPublicOpsIncGen
+
+	LINK_LIBS PUBLIC
```
Comment (benvanik) [inline llvm-projects/iree-dialects/lib/Dialects/iree_public/CMakeLists.txt:11]: tabs?

File: llvm-projects/iree-dialects/CMakeLists.txt (hunk 1 above)
Comment (stellaraccident) [reply llvm-projects/iree-dialects/CMakeLists.txt:4]: fixed (copypasta)

**Comment by stellaraccident** [comment]
PTAL

**Review by benvanik — APPROVED** [review]

**Pull request merged**